### ISSUE TITLE
Considerations on cloud-based encoding, packaging and origin workflows

### ISSUE THREAD

**Issue by acbegen** [ietf-wg-mops/draft-ietf-mops-streaming-opcons#81]
The encoder sync workshop revealed lots of problems and unfortunately a little solution to synchronized encoding and packaging that are essential in large scale events. 

Should we touch on these topics in this draft as they seem to be quite related to "operational considerations", mostly in the sense that we raise awareness instead of listing/proposing solutions.

Thoughts?

**Comment by SpencerDawkins** [comment]
@squarooticus and @ldaigle - this is the pointer to the workshop that @acbegen mentioned in this issue. It was quite good. 

On our editors call yesterday, we talked about whether this is material that should be in 
draft-ietf-mops-streaming-opcons. We were leaning toward "yes", but also recognizing that this is new material that the working group hasn't discussed (so not likely we can get to WGLC by Monday, when the I-D cutoff happens). 

I asked if this was a candidate for a second I-D, focused on encoder synchronization, but we couldn't think of a good reason NOT to include this topic in draft-ietf-mops-streaming-opcons, if we were going to document it in MOPS. 

You two could reasonably make a decision about that, or bring this up on the mailing list to get feedback from the working group, or (probably several other alternatives). Can we defer to the chairs for guidance about this?

Best,

Spencer

**Comment by squarooticus** [comment]
Sorry for the late reply: somehow I missed the mention in my inbox.

I don't see any reason why this should be out of scope. If you have time, I'd say add some proposed text prior to the cutoff later today and then post a summary of this to the list. We can raise the question at the 111 session and do a consensus call if there are any objections.

**Comment by gitnnelg** [comment]
I agree with Kyle that it’s not out of scope, but I do suggest that we attempt to be thoughtful of keeping the focus on IETF related aspects – networks, routing, network & transport protocols, addressing – so as to avoid wading too deep into the higher level aspects of Encoding and Packaging.

I guess what I’m trying to say, is conscious effort is needed to avoid getting into weeds that don’t directly connect with the IETF’s technical interests.  If we did go into the weeds, we’d be covering material that won’t connect with the IETF reader and might even leave them behind.

-glenn

On 7/12/21, 9:35 AM, "Kyle Rose" ***@***.******@***.***>> wrote:


Sorry for the late reply: somehow I missed the mention in my inbox.

I don't see any reason why this should be out of scope. If you have time, I'd say add some proposed text prior to the cutoff later today and then post a summary of this to the list. We can raise the question at the 111 session and do a consensus call if there are any objections.

**Comment by squarooticus** [comment]
That's a very good point, Glenn. If this has implications for transport, network QoS, scalability, etc., then it may be relevant.

**Comment by SpencerDawkins** [comment]
Re: IETF-related aspects, we should remember that we do have IETF working groups that touch on this topic, even if they are in ART, so I wouldn't worry about that. 

So, after thinking about comments from others, my other question is how much of the material we are talking about is well-understood and roughly agreed upon. Most of what we've got in the current version of the draft is either well-understood or at least has consensus behind it (example - the material about QUIC isn't as mature as the material about TCP, but I think it's mature enough). I might think that the point of having the Synchronization workshop was that we don't have as much of a handle on that yet. Unless we do understand that well enough to provide input about potential new work in the IETF, maybe it IS worth holding off on this, and doing a separate document when we can be confident. 

But this is definitely worth putting in front of the working group at IETF 111, no matter what the working group decides.

**Comment by squarooticus** [comment]
Sounds like a plan! 🐿

**Comment by acbegen** [comment]
Pending on the WG's decision, my suggestion would be to at least mention this topic and make a recommendation along the lines of: “this is important if you are doing linear content, so watch this space for developments”. I don't think we are in a position to say anything else beyond that because the work is still early at this stage.

**Comment by SpencerDawkins** [comment]
@acbegen, I agree, and I added your suggestion to the slides for IETF 111 MOPS. Thank you for making it!

**Comment by SpencerDawkins** [comment]
Insert this material in "For further reading" (#14}?

**Comment by acbegen** [comment]
This is now addressed in the reading list section. Issue can be closed.

**Comment by SpencerDawkins** [comment]
I agree that closing this issue is the right thing to do. I'd love to see us work on a "volume 2" in a year or two, and include topics like this one, but speculating about what might happen, or holding this document up, are both bad ideas (tm).
